app/scripts/models/import.js

```diff
@@ -1,178 +1,219 @@
-angular.module('wcagReporter')
-  .factory(
-    'wcagReporterImport',
-    function ($rootScope, evalModel, currentUser, reportStorage, importV1, changeLanguage) {
-      var jsonld = window.jsonld;
-
-      function objectCollide (obj1, obj2) {
-        Object.keys(obj1)
-          .forEach(function (prop) {
-            if (typeof obj1[prop] !== 'function' &&
-                typeof obj2[prop] !== 'undefined') {
-              obj1[prop] = obj2[prop];
-            }
-          });
-      }
-
-      function compactEach (callback) {
-        var testCallback;
-        var results = [];
-        var calls = 0;
-        var evalType = evalModel.context['@vocab'] + evalModel.type;
-        var personType = currentUser['@context']['@vocab'] + currentUser.type;
-
-        testCallback = function (err, compacted) {
-          results.push(compacted);
-          if (results.length === calls) {
-            callback(results);
-          }
-        };
-
-        return function (evalObj) {
-          calls += 1;
-
-          if (evalObj['@type'] &&
-                    evalObj['@type'].indexOf(evalType) !== -1) {
-            // Compact with the evaluation context
-            jsonld.compact(
-              evalObj,
-              evalModel.context,
-              testCallback
-            );
-          } else if (evalObj['@type'] &&
-                    evalObj['@type'].indexOf(personType) !== -1) {
-            // Compact with the FOAF context
-            jsonld.compact(
-              evalObj,
-              currentUser['@context'],
-              testCallback
-            );
-          } else {
-            results.push(evalObj);
+angular
+  .module('wcagReporter')
+  .factory('wcagReporterImport', function (
+    $rootScope,
+    evalModel,
+    currentUser,
+    reportStorage,
+    importV1,
+    changeLanguage
+  ) {
+    var jsonld = window.jsonld;
+
+    /**
+     * OBJECT MODIFIER
+     * Add to or replace object 1's keys with object 2's keys
+     * @param  {Object} obj1 Object that needs to be updated
+     * @param  {Object} obj2 Object with keys that need to replace or be added to obj1
+     * @return {undefined}      Modifies object 1 with object 2 keys
+     */
+    function objectCollide (obj1, obj2) {
+      Object.keys(obj1)
+        .forEach(function (prop) {
+          if (
+            typeof obj1[prop] !== 'function' &&
+            typeof obj2[prop] !== 'undefined'
+          ) {
+            obj1[prop] = obj2[prop];
           }
-        };
-      }
+        });
+    }
 
-      /**
+    function compactEach (callback) {
+      var testCallback;
+      var results = [];
+      var calls = 0;
+      var evalType = evalModel.context['@vocab'] + evalModel.type;
+      var personType = currentUser['@context']['@vocab'] + currentUser.type;
+
+      testCallback = function (err, compacted) {
+        if (err) {
+          console.error(err);
+        }
+
+        results.push(compacted);
+
+        if (results.length === calls) {
```

Comment:
so based on what I am reading, I saw `calls` as a simple counter to ensure when to return out of this callback?

I am sure we can write this in a better way.

Because what happens if we do `calls++` somewhere else and cause a side effect?

Changing `results.length >= calls` would make it slightly better, but I do not think that is any good.

Promises/ async/await may come to rescue here.